src/main/java/de/retest/web/RecheckSeleniumAdapter.java

```diff
@@ -49,19 +51,40 @@ public RecheckSeleniumAdapter() {
 
 	@Override
 	public boolean canCheck( final Object toVerify ) {
-		return toVerify instanceof WebDriver;
+		return toVerify instanceof WebDriver || toVerify instanceof RemoteWebElement //
```

Comment:
Just a short note from my side: We currently mix `WebDriver` and `RemoteWebElement` in the public API. Why not have both `WebDriver` and `WebElement` or `RemoteWebDriver` and `RemoteWebElement`?

Reply:
Well, the clean solution would be to have a adapter each (i. e. driver and element separated). The only part of the RemoteElement is the `WrapsDriver` interface (of which the driver is hopefully a `JavaScriptExecutor`). So technically we would need a check for `instanceof WebElement & WrapsDriver`. 

My implementation does this and provides a fallback to gather the information directly from the element.

However, as we want(/need) this out as fast as possible for the release, I settled with this (not so clean) implementation.

Reply:
>as we want(/need) this out as fast as possible for the release

It is public API, so I am pro investing some more time to come up with a proper solution. Once it's merged, it's gonna stay and we have to maintain it. `WebDriver` and `RemoteWebElement` just doesn't feel right.

In addition, many people use the `WebDriver` inteface on the left-hand side as their type, just have a look at the official Selenium API introduction:

https://www.seleniumhq.org/docs/03_webdriver.jsp#introducing-the-selenium-webdriver-api-by-example

This will only return `WebElement`s, so that wouldn't work with this API without casting. In contrast, `RemoteWebElement` implements `WebElement`, so this signature would be compatible.

Reply:
I agree that we should loosen the `RemoteWebElement` to a `WebElement` and check similar to: [extract](https://github.com/retest/recheck-web/blob/6d724b7123aeb356d4180cefd35056ab9822512e/src/main/java/de/retest/web/adapter/WebElementRecheckAdapter.java#L52) and [execute](https://github.com/retest/recheck-web/blob/6d724b7123aeb356d4180cefd35056ab9822512e/src/main/java/de/retest/web/mapping/JavaScriptWebDataProvider.java#L115).

That means, we only work with `WebElement`, `WrapsElement` and `WrapsDriver`.